{"name": "删除批量创建而产生的部门", "status": "failed", "statusDetails": {"message": "AssertionError: 断言失败 -|- 实际结果:部门下有子部门或成员 || 预期结果: 成功", "trace": "cls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '210008', 'errmsg': '部门下有子部门或成员'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n>               assert actual == v\nE               AssertionError\n\ntools/data_process.py:118: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\ncases = ['case_046', '测试数据清理', '成员，部门相关', '删除批量创建而产生的部门', '/v1/dept/delete?access_token=&$.case_001.access_token&', '', ...]\n\n    @pytest.mark.flaky(reruns=0, reruns_delay=1)\n    def test_main(cases):#cases, get_db\n        # 此处的cases入参来自与 conftest.py  文件中 cases函数，与直接使用 @pytest.mark.parametrize\n        # 有着差不多的效果\n        # 发送请求\n        response, expect, sql = BaseRequest.send_request(cases)\n        # 执行sql\n        # DataProcess.handle_sql(sql, get_db)\n        # 断言操作\n>       DataProcess.assert_result(response, expect)\n\ntest/test_api.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '210008', 'errmsg': '部门下有子部门或成员'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n                assert actual == v\n            except AssertionError:\n>               raise AssertionError(f'断言失败 -|- 实际结果:{actual} || 预期结果: {v}')\nE               AssertionError: 断言失败 -|- 实际结果:部门下有子部门或成员 || 预期结果: 成功\n\ntools/data_process.py:120: AssertionError"}, "steps": [{"name": "请求地址", "status": "passed", "attachments": [{"name": "请求地址", "source": "d17962a5-589b-4679-a2d8-05e604bea4c0-attachment.txt", "type": "text/plain"}], "start": 1622542331598, "stop": 1622542331599}, {"name": "请求头", "status": "passed", "attachments": [{"name": "请求头", "source": "687cc953-e764-42d9-99d1-8f9a7c6070c2-attachment.txt", "type": "text/plain"}], "start": 1622542331599, "stop": 1622542331599}, {"name": "请求参数", "status": "passed", "attachments": [{"name": "请求参数", "source": "aa6e7a88-a19c-469e-b400-5dac82a32a20-attachment.txt", "type": "text/plain"}], "start": 1622542331599, "stop": 1622542331599}, {"name": "上传文件", "status": "passed", "attachments": [{"name": "上传文件", "source": "fe092ad4-4956-4690-90de-e9796d0418c8-attachment.txt", "type": "text/plain"}], "start": 1622542331599, "stop": 1622542331600}, {"name": "响应耗时(s)", "status": "passed", "attachments": [{"name": "响应耗时(s)", "source": "223b3eb0-a391-4aee-bd39-c94e1ae03e2b-attachment.txt", "type": "text/plain"}], "start": 1622542331861, "stop": 1622542331861}, {"name": "响应内容", "status": "passed", "attachments": [{"name": "响应内容", "source": "10225ee3-5ae0-45d9-a781-55d275a9eab7-attachment.txt", "type": "text/plain"}], "start": 1622542331862, "stop": 1622542331862}, {"name": "存储实际响应", "status": "passed", "attachments": [{"name": "存储实际响应", "source": "e90f6b44-4c51-4917-a6e7-9beeaabe91ac-attachment.txt", "type": "text/plain"}], "start": 1622542331863, "stop": 1622542331865}, {"name": "第1个断言", "status": "passed", "attachments": [{"name": "第1个断言", "source": "6b2ff5d4-7e55-439e-8f29-9da32605c556-attachment.txt", "type": "text/plain"}], "start": 1622542331865, "stop": 1622542331866}], "parameters": [{"name": "cases", "value": "['case_046', '测试数据清理', '成员，部门相关', '删除批量创建而产生的部门', '/v1/dept/delete?access_token=&$.case_001.access_token&', '', 'get', 'params', '', '{\\n \"did\": \"&$.case_045.datas[1].did&\"\\n}', '', '', '{\"$.errmsg\":\"成功\"}']"}], "start": 1622542331598, "stop": 1622542331866, "uuid": "d81729db-43fb-42e0-a5af-de1e20d1d58f", "historyId": "fabce75fa8cda793c74aba4ae419210e", "testCaseId": "881516e7ef2e2aa5436f6a3c3e6a5bcb", "fullName": "test.test_api#test_main", "labels": [{"name": "tag", "value": "flaky(reruns=0, reruns_delay=1)"}, {"name": "parentSuite", "value": "test"}, {"name": "suite", "value": "test_api"}, {"name": "host", "value": "liuchengxudeMacBook-Pro.local"}, {"name": "thread", "value": "19614-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test.test_api"}]}